README.md

```diff
@@ -850,11 +850,24 @@ scenario where you expect the input to be valid most of the time.
  *
  * @param input         the Latin1 string to convert
  * @param length        the length of the string in bytes
- * @param latin1_output  the pointer to buffer that can hold conversion result
+ * @param utf8_output  the pointer to buffer that can hold conversion result
  * @return the number of written char; 0 if conversion is not possible
  */
 simdutf_warn_unused size_t convert_latin1_to_utf8(const char * input, size_t length, char* utf8_output) noexcept;
 
+/**
+ * Convert Latin1 string into UTF8 string with output limit.
+ *
+ * This function is suitable to work with inputs from untrusted sources.
+ *
+ * @param input         the Latin1 string to convert
+ * @param length        the length of the string in bytes
+ * @param utf8_output  	the pointer to buffer that can hold conversion result
+ * @param utf8_len  the maximum output length
+ * @return the number of written char; 0 if conversion is not possible
+ */
+simdutf_warn_unused size_t convert_latin1_to_utf8_safe(const char * input, size_t length, char* utf8_output, size_t utf8_len) noexcept;
```

Comment:
may I suggest keeping the name but changing the signature to take a std::span as output? that means the type system conveys the meaning. std span is C++20 which may be a no go.

Reply:
> std span is C++20 which may be a no go.

We support C++11. We could add `std::span` support, but it would have to be optional (so, on top of the existing API). Note that this applies to our whole API. I will add an issue.

Reply:
Done : https://github.com/simdutf/simdutf/issues/557